### PR TITLE
Add BlackPearl configuration to bendo server

### PR DESCRIPTION
**Pull request by dbrower**
This adds a `blackpearl://` and `blackpearls://` schemas for configuring bendo to use a remote BlackPearl tape device.

A lot of configuration that was happening in the `server` module was moved to the `cmd/bendo`. This way cmd/bendo can directly setup things using the config file and environment variables.

Documentation is updated with the new options.

### CODE REVIEW

**Review by msuhovec — COMMENTED** [review]
So is the DS3_TEMPDIR there because you can't stream uploads anymore?

File: cmd/bendo/location_test.go [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+		{"/abs/path/", "", typeFileSystem, "", ""},
+		{"file:/rel/path", "", typeFileSystem, "", ""},
+		{"file:rel/path", "", typeFileSystem, "", ""},
+		{"s3:/bucket", "", typeS3, "bucket", ""},
+		{"s3:/bucket", "more", typeS3, "bucket", "more/"},
+		{"s3://localhost:9000/bucket/prefix/", "", typeS3, "bucket", "prefix/"},
+		{"s3://localhost:9000/bucket/prefix/", "more", typeS3, "bucket", "prefix/more/"},
+		{"blackpearl:/bucket", "", typeBlackPearl, "bucket", ""},
+		{"blackpearl:/bucket", "more", typeBlackPearl, "bucket", "more/"},
+		{"blackpearl://192.168.1.70:9000/bucket/prefix/", "", typeBlackPearl, "bucket", "prefix/"},
```
Comment (msuhovec) [inline cmd/bendo/location_test.go:66]: No a big deal - why this default value? It's never gonna work, right?

**Review by msuhovec — APPROVED** [review]

**Comment by dbrower** [comment]
Yeah, the BlackPearl does not support streaming uploads since it requires you to give the size of the item in advance. So for now we save each item to disk and then upload it. I add this configuration since sometimes the default temp filesystem is in-memory only so we may want/need to change where we store the temp files.

File: cmd/bendo/location_test.go (hunk 1 above)
Comment (dbrower) [reply cmd/bendo/location_test.go:66]: This test is for parsing the string. It probably wouldn't work for connecting to it.

**Pull request merged**